docs/content/sips/012-signing-spin-releases.md

```diff
@@ -0,0 +1,206 @@
+title = "SIP 012 - "Signing the Spin releases"
+template = "main"
+date = "2023-03-01T01:01:01Z"
+
+---
+
+Summary: This improvement proposal describes the reasoning and implementation
+for signing Spin releases.
+
+Owners: radu@fermyon.com
+
+Created: March 2, 2023
+
+## Background
+
+Signing release artifacts for software offers a way for consumers of the software
+to verify the integrity of the package they downloaded. The signature should
+offer two guarantees: the author of the artifact is indeed the one expected by
+the consumer, and the content of the artifact has not been tampered with since
+its creation.
+
+The goal of this SIP is to define the tools used for generating signatures for
+the Spin project, and to provide a standard process for signing release artifacts
+for Spin and related projects.
+
+## Proposal
+
+This SIP proposes that for every [Spin release](https://github.com/fermyon/spin/releases),
+we add metadata that helps a user validate the integrity of the package they
+download.
+Specifically, this proposal suggests the use of [Sigstore](https://www.sigstore.dev/),
+_a new standard for signing, verifying, and protecting software_, and in particular,
+the use of [the new release of Cosign v2.0](https://blog.sigstore.dev/cosign-2-0-released),
+which stabilizes [keyless signatures using an OIDC provider](https://docs.sigstore.dev/cosign/keyless/).
+
+In keyless mode, Sigstore makes an OIDC provider the root of trust for the
+signature by creating short-lived x509 certificates bound to an OIDC identity.
+The certificates are authenticated and auditable, which makes the resulting
+signatures auditable as well.
+
+This process uses a few central pieces that are assumed trusted:
+
+- an OIDC provider (such as GitHub)
+- Fulcio — free root certificate authority that issues temporary certificates
+bound to an OIDC identity, published to Rekor
+- Rekor — transparency and timestamp service, provides a ledger that can be
+audited
+- `cosign sign` — the CLI that connects an OIDC identity to Fulcio to generate the
```

Comment:
Trying to get an idea of the external services involved when invoking this command...  Fulcio?  Others?

Perhaps a rare event but what is our approach when we're unable to sign due to external service issues?

Reply:
Fulcio and Rekor are both public good services that this mechanism uses while performing a signature.

If either is down, our signature process will fail.
Given that fairly large projects such as Kubernetes have already adopted this process, and that this has been marked as stable, my assumption is that the services should be relatively stable.

While we should sign canary builds, it's only stable releases that we are asking users to trust. Given that, rare events when we would be unable to sign stable releases should be indeed _rare_.
(if our experience is different as we test this, happy to re-evaluate this)